src/spikeinterface/qualitymetrics/pca_metrics.py

```diff
@@ -955,11 +1005,9 @@ def pca_metrics_one_unit(args):
     pc_metrics = {}
     # metrics
     if "isolation_distance" in metric_names or "l_ratio" in metric_names:
-        try:
-            isolation_distance, l_ratio = mahalanobis_metrics(pcs_flat, labels, unit_id)
-        except:
-            isolation_distance = np.nan
-            l_ratio = np.nan
```

Comment:
what if they fail? IMO, we should keep the try except and maybe add a warning.

If I'm computing 10 quality metrics for 500 units and one pc metric fail, this will stop the computation, or am I reading it wrong?

Reply:
same for all other sklearn based metrics, sometimes the fitting fails and that's ok :)

Reply:
Ok. So, keep these but make sure the tests, which we know should give non-Nan results, give non-Nan results? That sounds good!